File: daq/slc/src/dqm/src/Histo1D.cc
#include "Histo1D.hh"

#include "DoubleArray.hh"

using namespace B2DQM;

Histo1D::Histo1D() throw() : Histo1()
{
  _data = new DoubleArray(0);
}

Histo1D::Histo1D(const Histo1D& h) throw() : Histo1(h)
{
  _data = new DoubleArray(h._data->size());
}

Histo1D::Histo1D(const std::string& name, const std::string& title, int nbinx, double min, double max)
throw() : Histo1(name, title, nbinx, min, max)
{
  _data = new DoubleArray(nbinx + 2);
}

Histo1D::~Histo1D() throw() {}

std::string Histo1D::getDataType() const throw()
{
  return std::string("H1D");
}

File: daq/slc/extra/javalib/css/org.csstudio.belle2.daq/src/org/belle2/daq/dqm/Histo1D.java
package org.belle2.daq.dqm;

public class Histo1D extends Histo1 {

    public Histo1D() {
        super();
        data = new DoubleArray(0);
    }

    public Histo1D(String name, String title, int nbinx, double xmin, double xmax) {
        super(name, title, nbinx, xmin, xmax);
        data = new DoubleArray(nbinx + 2);
    }

    @Override
    public String getDataType() {
        return "H1D";
    }

}
